libraries/ms-drivers/src/x86/spv1020_mppt.c

```diff
@@ -0,0 +1,67 @@
+#include "spv1020_mppt.h"
+#include "log.h"
+
+// This file implements an x86 approximation of the stm32-specific functionality.
+// See ../spv1020_mppt.c for the shared functions.
+
+// These are "reasonable" values of the inputs.
+#define FIXED_CURRENT 512  // middle of the 10-bit current range
+#define FIXED_VIN 720      // 70% of max (2^10), about the maximum power point
+#define FIXED_PWM 500      // a duty cycle of 50%
+#define FIXED_STATUS 0
+
+static bool s_is_shut = false;
+
+StatusCode spv1020_shut(SpiPort port) {
+  if (port >= NUM_SPI_PORTS) {
+    return STATUS_CODE_INVALID_ARGS;
+  }
+
+  // this doesn't occur on stm32 but is useful for debugging
+  if (s_is_shut) {
+    LOG_WARN("Issuing SHUT command to SPV1020 MPPT when it's already shut\r\n");
```

Comment:
Why carriage returns?

Reply:
Minicom needs carriage returns to move to the beginning of the new line like a typewriter. There's a setting to auto-insert carriage returns, but that was the rational behind putting them in. (I suppose this is x86 so we aren't using minicom anyways, I'll remove them.)